pkg/kubelet/cm/deviceplugin/manager.go

```diff
@@ -283,46 +253,61 @@ func (m *ManagerImpl) Allocate(node *schedulercache.NodeInfo, attrs *lifecycle.P
 }
 
 // Register registers a device plugin.
-func (m *ManagerImpl) Register(ctx context.Context, r *pluginapi.RegisterRequest) (*pluginapi.Empty, error) {
-	glog.Infof("Got registration request from device plugin with resource name %q", r.ResourceName)
-	metrics.DevicePluginRegistrationCount.WithLabelValues(r.ResourceName).Inc()
-	if r.Version != pluginapi.Version {
-		errorString := fmt.Sprintf(errUnsupportedVersion, r.Version, pluginapi.Version)
-		glog.Infof("Bad registration request from device plugin with resource name %q: %v", r.ResourceName, errorString)
-		return &pluginapi.Empty{}, fmt.Errorf(errorString)
+func (m *ManagerImpl) register(r *pluginwatcher.PluginInfo) {
+	glog.Infof("got r.SockPath %s", r.SockPath)
+	if !strings.Contains(r.SockPath, ".sock") {
+		return
 	}
-
-	if !v1helper.IsExtendedResourceName(v1.ResourceName(r.ResourceName)) {
-		errorString := fmt.Sprintf(errInvalidResourceName, r.ResourceName)
-		glog.Infof("Bad registration request from device plugin: %v", errorString)
-		return &pluginapi.Empty{}, fmt.Errorf(errorString)
+	sock := strings.Split(r.SockPath, "/")
+	sock = strings.Split(sock[len(sock)-1], ".")
+	if len(sock) != 3 {
+		// NOTE: socket path must be in the form of /var/lib/kubelet/device-plugins/<domain-name>.<resource-name>.sock
```

Comment:
Let's have the name be explicit in the gRPC protocol rather than try to fit that into the FS naming convention.